### PR TITLE
7.0 worked days from timesheet payslip run integration

### PR DESCRIPTION
**Pull request by ghost**
This PR adds the possibility in payslip batch to import worked days from timesheet. This adds a simple box to check in the wizard form if the user wants to use this functionnality.

Also, I moved the yml test to python unittest. The reason for this is that unittest is allows to create data with loops, which does not seem to work in yml. This makes the tests easier to read and understand.

For the method get_timesheets_from_employee, it is a former section of the method import_worked_days that I separated so that it could be called by the payslip run wizard (hr_payslip_employees).

### CODE REVIEW

File: hr_worked_days_from_timesheet/__openerp__.py [hunk 1]
```diff
@@ -26,7 +26,9 @@
     'description': """
 Worked Days From Timesheet
 ==========================
-* Adds a button to import worked days from timesheet
+* Adds a button to import worked days from timesheet in a single payslip
+* Adds the option to import worked days from timesheet in payslip batches
+  when generating payslips.
```
Comment (bwrsandman) [inline hr_worked_days_from_timesheet/__openerp__.py:31]: Please use present imperative ("Add a button") if you don't use "This module" as a subject

File: hr_worked_days_from_timesheet/hr_payslip.py [hunk 2]
```diff
@@ -68,48 +68,85 @@ def timesheet_mapping(
...
+                cr, uid,
+                [('code', '=', code)],
+                context=context
+            )
+            date_format = lang_pool.read(
+                cr, uid,
+                lang_id,
+                ['date_format'],
+                context=context
+            )[0]['date_format']
```
Comment (bwrsandman) [inline hr_worked_days_from_timesheet/hr_payslip.py]: Get the pools and date formats outside of the loop so they are only fetched once for multiple ids

File: hr_worked_days_from_timesheet/i18n/fr.po [hunk 3]
```diff
@@ -59,15 +59,34 @@ msgid "Pay Slip"
...
+#: code:addons/hr_worked_days_from_timesheet/hr_payslip_employees.py:62
+#, python-format
+msgid "Warning!"
+msgstr "Attention!"
+
+#. module: hr_worked_days_from_timesheet
+#: code:addons/hr_worked_days_from_timesheet/hr_payslip_employees.py:63
+#, python-format
+msgid "You must select employee(s) to generate payslip(s)."
+msgstr "Vous devez sélectionner au moins un employé pour générer un(des) payslip(s)"
```
Comment (gurneyalex) [inline hr_worked_days_from_timesheet/i18n/fr.po]: don't we have a better translation in french for payslip ?

File: hr_worked_days_from_timesheet/i18n/fr.po [hunk 4]
```diff
@@ -59,15 +59,34 @@ msgid "Pay Slip"
...
 #. module: hr_worked_days_from_timesheet
 #: model:ir.model,name:hr_worked_days_from_timesheet.model_hr_payslip_worked_days
 msgid "Payslip Worked Days"
 msgstr "Jours travaillés"
+
+#. module: hr_worked_days_from_timesheet
+#: code:addons/hr_worked_days_from_timesheet/hr_payslip_employees.py:62
+#, python-format
+msgid "Warning!"
+msgstr "Attention!"
```
Comment (gurneyalex) [inline hr_worked_days_from_timesheet/i18n/fr.po]: missing space before '!'

**Comment by gurneyalex** [comment]
minor remarks apart, :+1:

**Comment by ghost** [comment]
Hi @rschnapka, I made a pull request that improves the hr_worked_days_from_timesheet. It adds the feature in payslip batches. Since you seem to use the module, I wonder if you could review my code so the PR can be merged. Thanks!

**Comment by rschnapka** [comment]
Addition sounds interesting. I will have a look at it ASAP. But #1 I am not really a coder/programmer and #2 I currently only use timesheets in ODOO-8.0. I will setup a 7.0-dev-system an have a look at your code though. It will just probably not be done just today...

**Comment by rschnapka** [comment]
Hello David, installed the module. Works out of the box. 
Only tested with simple one-liner examples as I have got no production-data or the like. Seems to do the expected... coding looks fine to me - but that is (concerning policies etc) something an experienced programmer should have a look at. My first impression at least is "good to go". Thanks!

**Comment by ghost** [comment]
@max3903, @bwrsandman I would need a review for this PR.

**Comment by ghost** [comment]
@rschnapka Thanks. Your help is much appreciated, I think functionnal reviews are even more import than technical reviews.

**Comment by max3903** [comment]
:+1:

**Pull request merged**